pilot/pkg/model/extensions.go

```diff
@@ -126,7 +128,7 @@ func matchPorts(portSelectors []*typeapi.PortSelector, port int) bool {
 	return false
 }
 
-func convertToWasmPluginWrapper(originPlugin config.Config) *WasmPluginWrapper {
+func convertToWasmPluginWrapper(originPlugin config.Config, secretAllowed ResourceReferenceAllowed) *WasmPluginWrapper {
```

Comment:
What about passing PushContext directly here, instead of passing a function.

Reply:
That function was added to test this easily without building lot of push context stuff

Reply:
Yes. the test would be easy, but, IMHO, it seems to be hard to read the codes.

Reply:
Alternatively, I think we may need to enrich the documentation of `ResourceReferenceAllowed` type.

Reply:
I will update the docs to make it more clear